Eventy/src/app/app-routing.module.ts

```diff
@@ -18,25 +18,40 @@ import {EditUserComponent} from './user-management/edit-user/edit-user.component
+import {AuthGuard} from './infrastructure/auth/auth.guard';
 
 
 const routes: Routes = [
   {path: '', component: HomeComponent},
-  {path: 'login', component: LoginComponent},
-  {path: 'register', component: RegisterComponent},
-  {path: 'organize-event', component: EventOrganizationComponent},
-  {path: 'add-service', component: AddServiceComponent},
-  {path: 'event-types', component: AllEventTypesComponent},
-  {path: 'add-event-types', component: CreateEventTypeComponent},
-  {path: 'edit-event-type/:id', component: EditEventTypeComponent},
-  {path: 'edit-service/:id', component: EditServiceComponent},
-  {path: 'my-services', component: MyServicesViewComponent},
-  {path: 'solution-categories', component: SolutionCategoryManagementComponent},
+  {path: 'login', component: LoginComponent, canActivate: [AuthGuard],
+    data: {role: []}},
+  {path: 'register', component: RegisterComponent, canActivate: [AuthGuard],
+    data: {role: []}},
+  {path: 'organize-event', component: EventOrganizationComponent, canActivate: [AuthGuard],
+    data: {role: ['Organizer']}},
+  {path: 'add-service', component: AddServiceComponent, canActivate: [AuthGuard],
+    data: {role: ['Provider']}},
+  {path: 'event-types', component: AllEventTypesComponent, canActivate: [AuthGuard],
+    data: {role: ['Admin']}},
+  {path: 'add-event-types', component: CreateEventTypeComponent, canActivate: [AuthGuard],
+    data: {role: ['Admin']}},
+  {path: 'edit-event-type/:id', component: EditEventTypeComponent, canActivate: [AuthGuard],
+    data: {role: ['Admin']}},
+  {path: 'edit-service/:id', component: EditServiceComponent, canActivate: [AuthGuard],
+    data: {role: ['Provider']}},
+  {path: 'my-services', component: MyServicesViewComponent, canActivate: [AuthGuard],
+    data: {role: ['Provider']}},
+  {path: 'solution-categories', component: SolutionCategoryManagementComponent, canActivate: [AuthGuard],
+    data: {role: ['Admin']}},
   {path: 'users/:id', component: OtherUserProfilePageComponent},
-  {path: 'profile/edit', component: EditUserComponent},
-  {path: 'profile', component: MyProfilePageComponent},
-  {path: 'create-reservation', component: CreateReservationComponent},
-  {path: 'fast-registration', component: FastRegistrationComponent},
+  {path: 'profile/edit', component: EditUserComponent, canActivate: [AuthGuard],
+    data: {role: ['Provider', 'Organizer', 'AuthenticatedUser', 'Admin']}},
+  {path: 'profile', component: MyProfilePageComponent, canActivate: [AuthGuard],
+    data: {role: ['Provider', 'Organizer', 'AuthenticatedUser', 'Admin']}},
+  {path: 'create-reservation', component: CreateReservationComponent, canActivate: [AuthGuard],
+    data: {role: ['Organizer']}},
+  {path: 'fast-registration', component: FastRegistrationComponent, canActivate: [AuthGuard],
+    data: {role: ['AuthenticatedUser']}},
   {path: '**', redirectTo: ''},
 ];
```

Comment:
Upgrade Profile is not here?

Reply:
Yeah that isn't there probably because that is the latest new addition? It probably will be added through the next merge!